mixer/pkg/attribute/envoyProtoBag.go

```diff
@@ -0,0 +1,277 @@
+	convertedStringMaps map[int32]attr.StringMap
+	stringMapMutex      sync.RWMutex
+
+	// to keep track of attributes that are referenced
+	referencedAttrs      map[attr.Reference]attr.Presence
+	referencedAttrsMutex sync.Mutex
+}
+
+// referencedAttrsSize is the size of referenced attributes.
+
+var envoyProtoBags = sync.Pool{
+	New: func() interface{} {
+		return &EnvoyProtoBag{
+			referencedAttrs: make(map[attr.Reference]attr.Presence, referencedAttrsSize),
+		}
+	},
+}
+
+func fillAddress(reqMap map[string]interface{}, address *core.Address, name string) {
+	reqMap[name+".ip"] = []byte(net.ParseIP(address.GetSocketAddress().GetAddress()).To16())
+	reqMap[name+".port"] = int64(address.GetSocketAddress().GetPortValue())
+}
+
+func reformatTime(theTime *timestamp.Timestamp, durationNanos int32) time.Time {
+	return time.Unix(theTime.Seconds, int64(theTime.Nanos)+int64(durationNanos))
+}
+
+// GetProtoBag returns a proto-based attribute bag.
+// When you are done using the proto bag, call the Done method to recycle it.
+func GetEnvoyProtoBagAuthz(req *authzGRPC.CheckRequest) *EnvoyProtoBag {
+	//, globalDict map[string]int32,	globalWordList []string) *EnvoyProtoBag {
+	pb := envoyProtoBags.Get().(*EnvoyProtoBag)
+
+	// build the message-level dictionary
+	reqMap := make(map[string]interface{})
+	//TODO account for other protocols
+	reqMap["context.protocol"] = "http"
+	reqMap["context.reporter.kind"] = "inbound"
+	fillAddress(reqMap, req.GetAttributes().GetDestination().GetAddress(), "destination")
+	fillAddress(reqMap, req.GetAttributes().GetSource().GetAddress(), "source")
+	reqMap["destination.principal"] = req.GetAttributes().GetDestination().GetPrincipal()
+	reqMap["source.principal"] = req.GetAttributes().GetSource().GetPrincipal()
+	reqMap["request.headers"] = attr.WrapStringMap(req.GetAttributes().GetRequest().GetHttp().GetHeaders())
+	reqMap["request.host"] = req.GetAttributes().GetRequest().GetHttp().GetHost()
+	reqMap["request.method"] = req.GetAttributes().GetRequest().GetHttp().GetMethod()
+	reqMap["request.path"] = req.GetAttributes().GetRequest().GetHttp().GetPath()
+	reqMap["request.scheme"] = req.GetAttributes().GetRequest().GetHttp().GetScheme()
+	reqMap["request.time"] = reformatTime(req.GetAttributes().GetRequest().GetTime(), 0)
+	reqMap["request.useragent"] = req.GetAttributes().GetRequest().GetHttp().GetHeaders()["user-agent"]
+
+	pb.reqMap = reqMap
+
+	scope.Debugf("Returning bag with attributes:\n%v", pb)
+
+	return pb
+}
+
+// GetProtoBag returns a proto-based attribute bag.
+// When you are done using the proto bag, call the Done method to recycle it.
+//num is the index of the entry from the message's batch to create a bag from
+func GetEnvoyProtoBagAccessLog(msg *accessLogGRPC.StreamAccessLogsMessage, num int) attr.Bag {
+	// build the message-level dictionary
+	pb := envoyProtoBags.Get().(*EnvoyProtoBag)
+	reqMap := make(map[string]interface{})
+	if httpLogs := msg.GetHttpLogs(); httpLogs != nil {
```

Comment:
https://www.envoyproxy.io/docs/envoy/latest/api-v2/data/accesslog/v2/accesslog.proto#envoy-api-msg-data-accesslog-v2-accesslogcommon 
Have not look through all the attributes, but here are the ones that I think we can add:
1) from upstream_cluster we could get destination_serivce_host and destination_service_name.
2) response flags could also be added.
3) from tls property, request sni and source/destination principals could be added.

Reply:
1 and 3 are implemented now. 2 I will work on.